src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+
+        ResponseEntity<List<ProductResponseDto>> response =
+                this.testRestTemplate.exchange(
+                        getBaseProductUrl(), HttpMethod.GET, null, new ParameterizedTypeReference<>() {
+                        });
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    void deleteProductSuccessfully() {
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<HttpStatus> response =
+                this.testRestTemplate.exchange(
+                        getProductUrl(Objects.requireNonNull(productResponseDto).getId()),
+                        HttpMethod.DELETE,
+                        null,
+                        HttpStatus.class);
+
+        assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+    }
+
+
+    @Nullable
+    private ProductResponseDto getProductResponseDto(ProductRequestDto productRequestDto) {
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.postForEntity(getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+
+        return response.getBody();
+    }
+
+    @NotNull
+    private static ProductRequestDto getProductRequestDto(ResourcesInUserResponseDto resourcesInUser, User user) {
+
+        List<ResourceQuantityRequestDto> listOfResourcesInProduct = getResourceInProductRequestDtos(resourcesInUser);
+
+        ProductRequestDto productRequestDto = new ProductRequestDto();
+        productRequestDto.setName("TestProduct");
+        productRequestDto.setAuthors(List.of("TestAuthors"));
+        productRequestDto.setDescription("Test");
+        productRequestDto.setOwnerId(user.getId());
+        productRequestDto.setSalePrice(50);
+        productRequestDto.setResourcesContent(listOfResourcesInProduct);
+
+        return productRequestDto;
+    }
+
+    @NotNull
+    private static List<ResourceQuantityRequestDto> getResourceInProductRequestDtos(ResourcesInUserResponseDto resourcesInUser) {
+        ResourceQuantityRequestDto resourceQuantityRequestDto = new ResourceQuantityRequestDto();
+        List<ResourceQuantityRequestDto> listOfResourcesInProduct = new ArrayList<>();
+        resourcesInUser.getResourcesAndQuantities().forEach(r ->
+        {
+            resourceQuantityRequestDto.setId(r.getResource().getId());
+            resourceQuantityRequestDto.setQuantity(5);
+            listOfResourcesInProduct.add(resourceQuantityRequestDto);
+        });
+        return listOfResourcesInProduct;
+    }
+
+    @Nullable
+    private ResourcesInUserResponseDto getResourcesInUserResponseDto(ResourceInUserRequestDto resourceInUserRequestDto) {
+        ResponseEntity<ResourcesInUserResponseDto> createResourceInUser =
+                this.testRestTemplate.postForEntity(getBaseResourceAvailabilityUrl(), resourceInUserRequestDto, ResourcesInUserResponseDto.class);
+
+        System.out.println(createResourceInUser.getBody().getOwner());
```

Comment:
Please, make sure we are not keeping any System.out in the code that gets merged.

Reply:
Done!